### ISSUE TITLE
Container with "io.rancher.container.network=true" no route to host

### ISSUE THREAD

**Issue by GameScripting** [rancher/rancher#10222]
**Rancher versions:**
rancher/server: v1.6.10
rancher/agent: v1.2.6

**Infrastructure Stack versions:**
healthcheck: v0.3.3
ipsec: rancher/net:v0.11.9 and rancher/net:holder@0b3e885efbd1
network-services: rancher/metadata:v0.9.4 and rancher/network-manager:v0.7.8
scheduler: rancher/scheduler:v0.8.2
kubernetes (if applicable): we are using cattle

**Docker version: (`docker version`,`docker info` preferred)**

```
Client:
 Version:      17.04.0-ce
 API version:  1.28
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Mon Apr  3 18:07:42 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.04.0-ce
 API version:  1.28 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Mon Apr  3 18:07:42 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

```
Containers: 58
 Running: 49
 Paused: 0
 Stopped: 9
Images: 63
Server Version: 17.04.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 400
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: 
containerd version: 422e31ce907fd9c3833a38d7b8fdd023e5a76e73
runc version: 9c2d8d184e5da67c95d601382adf14862e4f2228
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-96-generic
Operating System: Ubuntu 16.04.2 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 47.16GiB
Name: worker-node-2-prod
ID: KTJD:DWY4:R4LX:SHZ2:76EY:LQAZ:R7F7:XO7Z:V7UE:B52R:6RT2:S5LP
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support

```

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
```
~# cat /etc/os-release
NAME="Ubuntu"
VERSION="16.04.2 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.2 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
```
```
~# uname -r
4.4.0-96-generic
```

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
 Ubuntu 16.04.2 LTS (with KVM) (4.4.0), https://www.profitbricks.de/de/

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node rancher 

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Setup:**

We have a two node setup

Node 1) nginx reverse proxy to the outer world, on the managed network (call it `main-proxy`)
Node 2) `docker run`-deployed container with `io.rancher.container.network=true`, also running a nginx proxy (call it `rancher-docker-gateway`). App containers(s) setup up with docker-compose which are NOT part of the rancher managed network (call it `app-container`). The `rancher-docker-gateway` to connected to the `app-container`s docker network by calling `docker network connect`

Basic idea is: Traffic flows like `main-proxy` --> `rancher-docker-gateway` --> `app-container`
Reason we have this kind of setup is, we want network level isolation of the `app-container`s.

Basically this setup works fine most of the time. However "sometimes" we see a `no route to host` issue in the `main-proxy`. Only solution to fix the issue is to delete the `rancher-docker-gateway` (and recreate it of course) and rewire everything.

I was wondering why the `no route to host` seems to come right out of the nowhere. Is there any regular task running on ranchers side that checks/updates/refreshes/reconnects the IPsec/network stuff that might trigger the breaking of the routing?

**Comment by veered** [comment]
We are seeing the same error and are also using an Nginx reverse proxy w/ Rancher 1.6.10

**Comment by GameScripting** [comment]
After some digging around other `no route to host`-problems with rancher I think I found

1. a workaround
2. a problem and a solution

**Workarround**

Go to `stacks` --> `infrastructure` -->click `ipsec + 1 Sidekick` --> restart the hosts `rancher/net:holder`. Please not that restarting the whole `ipsec + 1 Sidekick` does not work, you have to do it one by one if necessary. For me it often where single hosts that lost connection.

**Solution**

Update Rancher to `v1.6.11`

The (this) issuse was already discussed in https://github.com/rancher/rancher/issues/9863. The the fix was discussed in https://github.com/rancher/rancher/issues/9971.
